header-rewrite-filter/header_processor.cc

```diff
@@ -543,6 +544,32 @@ namespace HeaderRewriteFilter {
     }
 }
 
+std::tuple<absl::Status, std::string> DynamicFunctionProcessor::getDynamicMetadata(Http::RequestOrResponseHeaderMap& headers, Envoy::StreamInfo::StreamInfo* streamInfo, absl::string_view key) {
+    try {
+        if (!streamInfo) {
+            return std::make_tuple(absl::InvalidArgumentError("Stream info is null"), "");
+        }
+
+        auto& request_metadata = streamInfo->dynamicMetadata();
```

Comment:
Can this be const? You'll have to make everything below it const.

I'm guessing not, guessing it doesn't return const objects correctly. But it'd be great if it works

Reply:
Just tried it out and it works! I also thought it wouldn't but since I'm accessing the value via `filter_metadata()` rather than `mutable_filter_metadata()` it makes sense that I can declare everything `const`.